contracts/core/Allo.sol

```diff
@@ -379,7 +379,7 @@ contract Allo is Transfer, Initializable, Ownable, AccessControl {
     /// @notice allocate to multiple pools
     /// @param _poolIds ids of the pools
     /// @param _datas encoded data unique to the allocation strategy for that pool
-    function batchAllocate(uint256[] calldata _poolIds, bytes[] memory _datas) external payable {
+    function batchAllocate(uint256[] calldata _poolIds, bytes[] memory _datas) external {
```

Comment:
I removed payable from `batchAllocate`, because when the `batchAllocate` function is called, it iterates over the `_allocate` function for each pool and sends the entire msg.value with each allocation. However, this approach does not work as intended.

I currently see two ways to deal with it:
- Remove the payable keyword and use batchAllocate only for erc20 token (done)
- Modify the code to allocate the appropriate amount of msg.value for each allocation call

@thelostone-mc @codenamejason @zobront

Reply:
> Remove the payable keyword and use batchAllocate only for erc20 token

Ah this seems to be the cleanest way to go about this. 
This problem would persist even if we introduced OZ multi-call right ?